pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -696,6 +697,29 @@ func (ssr *Snapshotter) snapshotEventHandler(stopCh <-chan struct{}) error {
 				}
 			}
 
+		case <-ssr.fullSnapshotLeaseUpdateTimer.C:
+			if ssr.healthConfig.SnapshotLeaseRenewalEnabled {
+				if ssr.PrevFullSnapshot != nil {
+					ctx, cancel := context.WithTimeout(leaseUpdateCtx, brtypes.LeaseUpdateTimeoutDuration)
+					if _, err := heartbeat.FullSnapshotCaseLeaseUpdate(ctx, ssr.logger, ssr.PrevFullSnapshot, ssr.K8sClientset, ssr.healthConfig.FullSnapshotLeaseName, ssr.healthConfig.DeltaSnapshotLeaseName); err != nil {
+						ssr.logger.Warnf("FullSnapshot lease update failed : %v", err)
+						ssr.logger.Infof("Resetting the FullSnapshot lease to retry updating with revision %d after %v", ssr.PrevSnapshot.LastRevision, fullSnapshotLeaseUpdateRetryInterval.String())
+						ssr.fullSnapshotLeaseUpdateTimer.Stop()
+						ssr.fullSnapshotLeaseUpdateTimer.Reset(fullSnapshotLeaseUpdateRetryInterval)
+					} else {
+						ssr.logger.Infof("FullSnapshot lease successfully updated with revision %d", ssr.PrevSnapshot.LastRevision)
+						ssr.logger.Infof("Stopping the FullSnapshot lease update")
+						ssr.fullSnapshotLeaseUpdateTimer.Stop()
+					}
+					cancel()
+				} else {
+					ssr.logger.Infof("Skipping the FullSnapshot lease update since no full snapshot has been taken yet")
+					ssr.logger.Infof("Resetting the FullSnapshot lease to retry updating after %v", fullSnapshotLeaseUpdateRetryInterval.String())
+					ssr.fullSnapshotLeaseUpdateTimer.Stop()
+					ssr.fullSnapshotLeaseUpdateTimer.Reset(fullSnapshotLeaseUpdateRetryInterval)
+				}
+			}
+
```

Comment:
Thanks for pointing this out Ishan. 

I chose this path because the earlier code made calls to `FullSnapshotCaseLeaseUpdate` function from the `snapshotEventHandler()` function itself, so went ahead with a similar view.

But I agree with your points on how it's meant only for managing snapshots and nothing else. I'll move this part of the code to `heartbeat.go` file as suggested. Thanks!

Reply:
Updated this in the latest commit. PTAL @ishan16696